src/main/java/dev/callumherr/modding/ducky_lib/block/entity/AbstractMultiBlockExteriorEntity.java

```diff
@@ -0,0 +1,62 @@
+package dev.callumherr.modding.ducky_lib.block.entity;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+import net.minecraft.world.level.block.state.BlockState;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Abstract base class for exterior block entities in a multi-block structure.
+ *
+ * <p>This class provides the foundation for exterior blocks within a multi-block structure,
+ * including methods for retrieving the core block entity of the structure.</p>
+ */
+public abstract class AbstractMultiBlockExteriorEntity extends BlockEntity {
```

Comment:
For exterior blocks we dont need block entities afaik. simply on use refer to the core block. Maybe also a good idea to create a MultiBlockController that you have to interact with instead? (If possible id like options of using controller or not)

Reply:
lol yeah i can do that , ext was to test coz how cna i click the core coz its inside